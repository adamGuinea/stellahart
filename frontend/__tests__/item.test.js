import ItemComponent from '../components/Item';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
    id: '424242',
    title: 'Nike Air',
    price: 300,
    description: 'This is a really cool item',
    image: 'shoe.jpg',
    largeImage: 'bigShoe.jpg',
};

xdescribe('<Item />', () => {
    it('renders and matches the snapshot', () => {
        const wrapper = shallow(<ItemComponent item=
            {fakeItem} />);
        
        expect(toJSON(wrapper)).toMatchSnapshot()
    });
    // it('renders and image properly', () => {
    //     const wrapper = shallow(<ItemComponent item=
    //     {fakeItem} />)
    //     const img = wrapper.find('img');
    //     expect(img.props().src).toBe(fakeItem.image);
    //     expect(img.props().alt).toBe(fakeItem.title);
    // })
    // it('renders pricetag and title properly', () => {
    //     const wrapper = shallow(<ItemComponent item=
    //         {fakeItem} />)
    //         const PriceTag = wrapper.find('PriceTag')
    //         expect(PriceTag.children().text()).toBe('$3');
    //         expect(wrapper.find('Title a').text()).toBe
    //         (fakeItem.title);
    // });
    // it('renders out the buttons properly', () => {
    //     const wrapper = shallow(<ItemComponent item=
    //     {fakeItem} />);
    //     const buttonList = wrapper.find('.buttonList');
    //     expect(buttonList.children()).toHaveLength(3);
    //     expect(buttonList.find('Link')).toBeTruthy();
    //     expect(buttonList.find('AddToCart')).toBeTruthy();
    //     expect(buttonList.find('DeleteItem')).toBeTruthy();
    // });
});