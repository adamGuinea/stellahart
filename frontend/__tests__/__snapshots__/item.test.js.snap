// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="Nike Air"
    src="shoe.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "424242",
          },
        }
      }
    >
      <a>
        Nike Air
      </a>
    </Link>
  </Title>
  <PriceTag>
    $3
  </PriceTag>
  <p>
    This is a really cool item
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "424242",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="424242"
    />
    <DeleteItem
      id="424242"
    >
      Delete This Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
